src/libraries/Common/src/Interop/Browser/Interop.Runtime.Bridge.cs

```diff
@@ -0,0 +1,369 @@
+                    case TypeCode.Single:
+                        res[c] = 'f';
+                        break;
+                    case TypeCode.Double:
+                        res[c] = 'd';
+                        break;
+                    case TypeCode.String:
+                        res[c] = 's';
+                        break;
+                    default:
+                        if (t == typeof(IntPtr))
+                        {
+                            res[c] = 'i';
+                        }
+                        else if (t == typeof(Uri))
+                        {
+                            res[c] = 'u';
+                        }
+                        else
+                        {
+                            if (t.IsValueType)
+                                throw new NotSupportedException("ValueType arguments are not supported.");
+                            res[c] = 'o';
+                        }
+                        break;
+                }
+            }
+            return new string(res);
+        }
+
+        internal static void SetupJSContinuation(Task task, JSObject continuationObj)
+        {
+            if (task.IsCompleted)
+                Complete();
+            else
+                task.GetAwaiter().OnCompleted(Complete);
+
+            void Complete()
+            {
+                try
+                {
+                    if (task.Exception == null)
+                    {
+                        object? result;
+                        Type task_type = task.GetType();
+                        if (task_type == typeof(Task))
+                        {
+                            result = Array.Empty<object>();
+                        }
+                        else
+                        {
+                            result = task_type.GetMethod("get_Result")?.Invoke(task, Array.Empty<object>());
+                        }
+                        continuationObj.Invoke("resolve", result);
+                    }
+                    else
+                    {
+                        continuationObj.Invoke("reject", task.Exception.ToString());
+                    }
+                }
+                catch (Exception e)
+                {
+                    continuationObj.Invoke("reject", e.ToString());
+                }
+                finally
+                {
+                    continuationObj.Dispose();
+                    FreeObject(task);
+                }
+            }
+        }
+
+        internal static string ObjectToString(object o)
+        {
+            return o.ToString() ?? string.Empty;
+        }
+
+        internal static double GetDateValue(object dtv)
+        {
+            if (dtv == null)
+                throw new ArgumentNullException(nameof(dtv), "Value can not be null");
+            if (!(dtv is DateTime dt))
+                throw new InvalidCastException($"Unable to cast object of type {dtv.GetType()} to type DateTime.");
```

Comment:
It'd be good to move such messages into the .resx.  If you don't want to do it now, an issue is probably worthwhile.

Reply:
Setup issue here https://github.com/dotnet/runtime/issues/36999